src/packages/core/src/servers/utils/fragment-generator.ts

```diff
@@ -0,0 +1,43 @@
+export function* getFragmentGenerator(
+  data: Generator<Buffer, void, void>,
+  bufferSize: number
+): Generator<Buffer> {
+  // Use a large buffer to reduce round-trips
+  let buf = Buffer.allocUnsafe(bufferSize);
+  let offset = 0;
+
+  for (const value of data) {
+    const { byteLength } = value;
+    if (offset > 0 && byteLength + offset > bufferSize) {
+      yield buf.subarray(0, offset);
+      // Reset the buffer. Since `uws` sends packets asynchronously,
+      // it is important that we allocate a new buffer for the next
+      // frame. This avoids overwriting data before it is sent. The
+      // reason we need to do this is likely because we do not yet
+      // handle backpressure. Part of handling backpressure will
+      // involve the drain event and only sending while
+      // `ws.getBufferedAmount() < ACCEPTABLE_BACKPRESSURE`.
+      // See https://github.com/trufflesuite/ganache/issues/FIX THIS
+      buf = null;
+      offset = 0;
+    }
+    // Store value in buffer if it fits (but don't store it if it is the exact
+    // same size as bufferSize)
+    if (byteLength < bufferSize) {
+      // copy from value into buffer
+      if (buf === null) buf = Buffer.allocUnsafe(bufferSize);
+      value.copy(buf, offset, 0, byteLength);
+      offset += byteLength;
+    } else {
+      // Cannot fit this value in buffer, send it directly.
+      // Buffer has just been flushed so we do not need to worry about
```

Comment:
I added this here: https://github.com/trufflesuite/ganache/pull/2634/files#diff-0a3bee479fd46bab9ab4aa0beb7492951a25bce519c7196daf63a587e94294e8R35-R38

Let me know if it works or if it can be improved!

Reply:
Love it ❤️ !